### PR TITLE
Create new admin page to edit competitions upper bound limit for max submission size

### PR DESCRIPTION
**Pull request by OhMaley**
# A brief description of the purpose of the changes contained in this PR.
Creation of an admin page to edit admin settings for competitions.
For now it allows admin to edit the upper bound limit for max submission size that an organizer can set.

# A checklist for hand testing
- [ ] Check that the page can only be access by admin
- [ ] Check that the edition is working
- [ ] Check that an organizer is allowed to change the max_submission_size when set to a value bellow the upper_bound_limit_for_max_submission_size
- [ ] Check that an organizer is NOT allowed to change the max_submission_size when set to a value higher than the upper_bound_max_submission_size
- [ ] Check that an error is displayed bellow the max_submission_size field in case of a wrong value

# Misc. comments
Tested locally

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: codalab/apps/web/views.py
Comment (Tthomas63) [inline codalab/apps/web/views.py:209]: I don't remember off the top of my head, but is `start_date` not a field on the competition itself? (Necessitating sorted/reversed)?

File: codalab/apps/web/templates/web/admin_competitions_manager.html
Comment (Tthomas63) [inline codalab/apps/web/templates/web/admin_competitions_manager.html:155]: Nitpick: No newline at EOF

File: codalab/apps/web/models.py
Comment (OhMaley) [reply codalab/apps/web/models.py:293]: The organizer can only set the max submission size to 0. It will work the same way as any other value. It does not seem to have any side effect from what I've tested

File: codalab/apps/web/tests/test_competitions.py
Comment (OhMaley) [reply codalab/apps/web/tests/test_competitions.py:158]: Yes: for the test `test_can_edit_competition_with_ownership` it need to have a ParticipantStatus.APPROVED created in the DB. What I could do as an alternative is to create another `ParticipantStatus` with a status `APPROVED` so it would be available during the test. But I looked at all the tests and the status PENDING did not seem to be used so I just changed it.
If I missed something let me know

File: codalab/apps/web/views.py
Comment (OhMaley) [reply codalab/apps/web/views.py:209]: Yes start_date is a field on the Competition itself.
What do you mean by "_Necessitating sorted/reversed_" ?

File: codalab/apps/web/tests/test_competitions.py
Comment (Tthomas63) [reply codalab/apps/web/tests/test_competitions.py:158]: This makes sense, I was more of wondering if it was passing before without the status set to approved and if that had changed.

File: codalab/apps/web/views.py
Comment (Tthomas63) [reply codalab/apps/web/views.py:209]: So if the `start_date` is not a field in the database. there would be no way to sort the initial query by that field (without some extra query magic).

> "Necessitating sorted/reversed"

Meaning since we cannot use queries to sort our data, sorted/reversed is used instead of a query:
```
list(reversed(sorted(competitions, key=lambda c: c.get_start_date)))
```

Comment (OhMaley) [reply codalab/apps/web/views.py:209]: Actually it seems that I can sort the competitions on the staret_date field using the query. I'm updating it

**Comment by Tthomas63** [comment]
Think this was the edge-case I hit testing:
```
django            | 2022-06-02 14:37:17,796 ERROR Internal Server Error: /admin_competitions_manager/
django            | Traceback (most recent call last):
django            |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 41, in inner
django            |     response = get_response(request)
django            |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 187, in _get_response
django            |     response = self.process_exception_by_middleware(e, request)
django            |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 185, in _get_response
django            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
django            |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 68, in view
django            |     return self.dispatch(request, *args, **kwargs)
django            |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 88, in dispatch
django            |     return handler(request, *args, **kwargs)
django            |   File "/app/codalab/apps/web/views.py", line 203, in get
django            |     return super(AdminCompetitionsManager, self).get(*args, **kwargs)
django            |   File "/usr/local/lib/python3.8/site-packages/extra_views/formsets.py", line 322, in get
django            |     return super(BaseModelFormSetView, self).get(request, *args, **kwargs)
django            |   File "/usr/local/lib/python3.8/site-packages/extra_views/formsets.py", line 285, in get
django            |     return self.render_to_response(self.get_context_data(formset=formset))
django            |   File "/app/codalab/apps/web/views.py", line 210, in get_context_data
django            |     competitions = list(reversed(sorted(competitions, key=lambda c: c.get_start_date)))
django            | TypeError: can't compare offset-naive and offset-aware datetimes
```

**Review by Tthomas63 — APPROVED** [review]
Looks good to me. Think my only thoughts are:
 - Address datetime edge case
 - Migrations. We have conflicting migration numbers for web between our PR's. I think we can potentially use `python manage.py makemigrations web --merge` in this case since you're adding a field. I've also been the hold-up so let me know what's most convenient

**Pull request merged**
